Semana_20_Testes_Unitarios/TestesUnitarios/TestesUnitarios/Avaliacao/Business/ServicoProcessoGuia.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Linq;
+using TestesUnitarios.Avaliacao.Dao;
+using TestesUnitarios.Avaliacao.Entidades;
+
+namespace TestesUnitarios.Avaliacao.Business
+{
+    public class ServicoProcessoGuia : IServicoProcessoGuia
+    {
+        private IServicoEnvioGuias _servicoGuias;
+        private IDao<IGuiaProperties> _daoBase;
+        private IDao<IDespesasGuiaProperties> _daoBaseDespesas;
+
+        public ServicoProcessoGuia(IServicoEnvioGuias servicoGuia, IDao<IGuiaProperties> daoBase, IDao<IDespesasGuiaProperties> daoBaseDespesas)
+        {
+            _servicoGuias = servicoGuia;
+            _daoBase = daoBase;
+            _daoBaseDespesas = daoBaseDespesas;
+        }
+
+        public ServicoProcessoGuia()
+            : this(new ServicoEnvioGuias(), new Dao<Guia, IGuiaProperties>(), new Dao<DespesasGuia, IDespesasGuiaProperties>())
+        {
+        }
+
+        public RespostaProcessamentoDto ProcessarGuia(Guia guia)
+        {
+            var repostaProcessamento = new RespostaProcessamentoDto();
+
+            if (ValidacoesGuia(guia))
```

Comment:
👎 O ideal seria inverter esses ifs para diminuir o aninhamento do código e facilitar a leitura, segue um link de referência https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting